modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -499,3 +530,563 @@ class ProtocolBaseController extends BaseEntityController
+		rowController = new EndpointController
+			el: tr
+			thingRef: @model
+			valueDefs: EndpointsValuesConf
+			stateType: 'metadata'
+			stateKind: 'data column order'
+			rowNumber: rowNumber
+			rowNumberKind: @rowNumberKind
+			readOnly: @options.readOnly
+		# Add this controller to our tracking dictionary so we can access it later
+		@endpointControllers[rowNumber] = rowController
+
+		#if time/units are disabled, remove cells
+		if window.conf.protocol.endpointManager.showTime == false
+			@$(".bv_timeUnitsPickListParent").remove()
+			@$(".bv_timeInputParent").remove()
+		
+		#if concentration/units are disabled, remove the cells
+		if window.conf.protocol.endpointManager.showConcentration == false
+			@$(".bv_concentrationUnitsPickListParent").remove()
+			@$(".bv_concentrationInputParent").remove()
+
+
+	
+	getRowNumberForState: (state) =>
+		rowValues = state.getValuesByTypeAndKind 'numericValue', @rowNumberKind
+		if rowValues.length == 1
+			return rowValues[0].get('numericValue')
+		else
+			return 0
+	
+	getNextRowNumber: =>
+		row_nums = @collection.map @.getRowNumberForState
+		if row_nums.length > 0
+			return Math.max(...row_nums) + 1
+		else
+			return 1
+
+	handleAddEndpointPressed: =>
+		# Create a new LsState
+		lsState = @model.get("lsStates").createStateByTypeAndKind "metadata", "data column order"
+		# Set column order value
+		rowNum = @.getNextRowNumber()
+		rowNumValue = lsState.getOrCreateValueByTypeAndKind 'numericValue', @rowNumberKind
+		rowNumValue.set("numericValue", rowNum)
+		# Add the state to the collection
+		@collection.push lsState
+		@.addOne(lsState)
+	
+	handleRowRemoved: (rowNumber) =>
+		if @endpointControllers[rowNumber]?
+			@endpointControllers[rowNumber].remove()
+			@endpointControllers[rowNumber].unbind()
+			@endpointControllers[rowNumber].el.remove()
+
+	#Function brought over from experiment.coffee 
+	setupExperimentSummaryTable: (experiments) =>
+		#@$(".bv_searchStatusIndicator").addClass "hide"
+		$(".bv_experimentTableController").removeClass "hide"
+		if window.conf.experiment?.mainControllerClassName? and window.conf.experiment.mainControllerClassName is "EnhancedExperimentBaseController"
+			experimentListClass = "EnhancedExperimentList"
+		else
+			experimentListClass = "ExperimentList"
+		@experimentSummaryTable = new ExperimentSummaryTableController
+			el: $(".bv_experimentTableController")
+			collection: new window[experimentListClass] experiments
+
+		@experimentSummaryTable.on "selectedRowUpdated", @selectedExperimentUpdated
+
+		@experimentSummaryTable.render()
+		
+		#TODO - should there be a title over the table with the endpoint that is being searched for? 
+		if experiments.length == 0
+			$(".bv_experimentTableController").empty() 
+			$(".bv_experimentTableController").append "There are no matching experiments using this protocol and endpoint."
+			$(".bv_downloadFiles").hide()
+		else
+			$(".bv_downloadFiles").show()
+	
+	
+	downloadFiles: => 
```

Comment:
The downloadFiles logic is all ported over from ExperimentBrowser.coffee (ACAS-321).